tests/integration/master.test.js

```diff
@@ -0,0 +1,255 @@
+    let val = u8a_hex(randomish_u8a());
+    let sudocall = nc.tx.sudo.sudo(nc.tx.system.setStorage([[key, val]]));
+    await sign_send_tx(sudocall);
+    let bs = (await nc.rpc.state.getStorage(key)).unwrap();
+    expect(u8a_hex(bs)).toEqual(val);
+  }, 20000)
+
+  test('Root call with no votes', async () => {
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    await master_set_storage(nc, key, val, []);
+    let sto = await nc.rpc.state.getStorage(key);
+    assert(sto.isNone, "storage item should not have been set");
+  }, 20000);
+
+  test('Root call with invalid votes', async () => {
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    const did_to_key = [
+      [ALICE_DID, await keypair(randomish_u8a())],
+      [CHARLIE_DID, await keypair(randomish_u8a())],
+    ];
+    await master_set_storage(nc, key, val, did_to_key);
+    let sto = await nc.rpc.state.getStorage(key);
+    assert(sto.isNone, "storage item should not have been set");
+  }, 20000);
+
+  test('Root call with valid votes', async () => {
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    const did_to_key = [
+      [ALICE_DID, await keypair(ALICE_SK)],
+      [CHARLIE_DID, await keypair(CHARLIE_SK)],
+    ];
+    await master_set_storage(nc, key, val, did_to_key);
+    let sto = await nc.rpc.state.getStorage(key);
+    let u8a = sto.unwrap();
+    expect(u8a_hex(u8a)).toEqual(val);
+  }, 20000);
+
+  test('Root call with valid votes but insufficient vote count', async () => {
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    const did_to_key = [
+      [ALICE_DID, await keypair(ALICE_SK)]
+    ];
+    await master_set_storage(nc, key, val, did_to_key);
+    let sto = await nc.rpc.state.getStorage(key);
+    assert(sto.isNone, "storage item should not have been set");
+  }, 20000);
+
+  test('Root call with valid votes and oversufficient vote count', async () => {
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    const did_to_key = [
+      [ALICE_DID, await keypair(ALICE_SK)],
+      [BOB_DID, await keypair(BOB_SK)],
+      [CHARLIE_DID, await keypair(CHARLIE_SK)],
+    ];
+    await master_set_storage(nc, key, val, did_to_key);
+    let sto = await nc.rpc.state.getStorage(key);
+    let u8a = sto.unwrap();
+    expect(u8a_hex(u8a)).toEqual(val);
+  }, 20000);
+
+  test('Root call with votes not sorted lexically', async () => {
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    const did_to_key = [
+      [BOB_DID, await keypair(BOB_SK)],
+      [ALICE_DID, await keypair(ALICE_SK)],
+    ];
+    await master_set_storage(nc, key, val, did_to_key);
+    let sto = await nc.rpc.state.getStorage(key);
+    let u8a = sto.unwrap();
+    expect(u8a_hex(u8a)).toEqual(val);
+  }, 20000);
+});
+
+// connect to running node
+async function connect() {
```

Comment:
`DockAPI`'s `init` does this so a call like `(new DockAPI()).init(FullNodeEndpoint)` would be equivalent.

Reply:
Thanks. I'm aware. This module intentionally does not depend on sdk (except for types.json).

Reply:
Whats the concern with using the SDK?

Reply:
Decoupling from SDK makes this test portable and self-contained. It might be best to eventually move e2e node tests over to the dock-substrate repo, in that case depending on the sdk would cause problems related to circular dependency.

Reply:
Makes sense